database/data_seeding/nosql_seeder.js

```diff
@@ -23,18 +23,18 @@ const seedNoSqlData = async (numRecords, pace, startingId) => {
   let currentBatch = 1;
 
   while (numRecords > 0) {
-    const data = generateData(pace, startingId, 'nosql');
+    const data = generateData(batchSize, startingId, 'nosql');
 
     await pf.bulk({ docs: data })
```

Comment:
I am not sure if "await" is needed here because you are following this line with a ".then" chain. I think the purpose of async/await is to basically eliminate .then blocks. I think these few lines might be able to be written more like this if you want to use await:

try{
  await pf.bulk({ docs: data })
  console.log(`Batch number ${currentBatch} / ${numBatches} successfully seeded for IDs ${startingId} through ${(startingId + batchSize) - 1}`);
} catch (error) {
  throw new Error(`Error seeding database on batch number ${currentBatch} / ${numBatches} for IDs ${startingId} through ${(startingId + batchSize) - 1}`);
}

Reply:
good point, didn't even realize that, good catch!